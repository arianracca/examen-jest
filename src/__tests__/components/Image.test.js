import React from "react";
import renderer from 'react-test-renderer';
import { render } from "@testing-library/react";
import Image from "../../components/Image";

describe("Image component", () => {
  test("renders correctly", () => {
    const url = "https://mocking.com/mock.jpg";
    const title = "Example Image";
    
    render(<Image url={url} title={title} />);
    const imageItem = document.querySelector("img");
    
    expect(imageItem).toBeInTheDocument();
    expect(imageItem.src).toBe(url);
    expect(imageItem.alt).toBe(title);
  });

  test('should match snapshot', () => {
    const props = {
      url: 'https://mocking.com/mock.jpg',
      alt: 'Mock Image',
    };

    const component = renderer.create(<Image {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
