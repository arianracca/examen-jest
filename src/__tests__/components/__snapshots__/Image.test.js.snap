// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should match snapshot 1`] = `
<li>
  <img
    src="https://mocking.com/mock.jpg"
  />
</li>
`;
